wolfHSM/src/chapter03.md

```diff
@@ -0,0 +1,228 @@
+posixTransportTcpConfig posixTransportTcpCfg = {
+    /* IP and port configuration */
+};
+
+/* Step 2: Allocate client comm configuration and bind to the transport */
+/* Configure the client comms to use the selected transport configuration */
+whCommClientConfig commClientCfg = {
+             .transport_cb      = posixTransportTcpCb,
+             .transport_context = (void*)posixTransportTcpCtx,
+             .transport_config  = (void*)posixTransportTcpCfg,
+             .client_id         = 123, /* unique client identifier */
+};
+
+/* Subsequent steps remain the same... */
+```
+
+Note that the echo request in step 6 is just a simple usage example. Once the connection to the server is set up, any of the client APIs are available for use.
+
+## Basic Server Configuration
+
+*Note: A wolfHSM port comes with a referenc
+e server application that is already configured to run on the HSM core and so manual server configuration is not required.*
+
+Configuring a wolfHSM server involves allocating a server context structure and initializing it with a valid client configuration that enables it to perform the requested operations. These operations usually include client communication, cryptographic operations, managing keys, and non-volatile object storage. Depending on the required functionality, not all of these configuration components need to be initialized.
+
+
+The steps required to configure a server that supports client communication, NVM object storage using the NVM flash configuration, and local crypto (software only) are:
+
+1. Initialize the server comms configuration
+    1. Allocate and initialize a transport configuration structure, context, and callback implementation for the desired transport
+    2. Allocate and initialize a comm server configuration structure using the transport configuration from step 1.1
+2. Initialize the server NVM context
+    1. Allocate and initialize a config, context, and callback structure for the low-level flash storage drivers (the implementation of these structures is provided by the port)
+    2. Allocate and initialize an NVM flash config, context, and callback strucure and bind the port flash configuration from step 2.1 to them
+    3. Allocate an NVM context structure and initialize it with the configuration from step 2.2 using `wh_Nvm_Init()`
+3.  Allocate and initialize a crypto context structure for the server
+4. Initialize wolfCrypt (before initializing the server)
+5. Allocate and initialize a server config structure and bind the comm server configuration, NVM context, and crypto context to it
+6. Allocate a server context structure and initialize it with the server configuration using `wh_Server_Init()`
+7. Set the server connection state to connected using `wh_Server_SetConnected()` when the underlying transport is ready to be used for client communication (see [TODO](todo) for more information)
+8. Process client requests using `wh_Server_HandleRequestMessage()`
```

Comment:
weird - the nested bullets are rendered as 1.a, 1.b, 1.c, despite them being written as 1.1, 1.2, 1.3. So referring to "step 1.1" in the text is broken